### PR TITLE
Cache search results for 1 minute. Should reduce load spikes.

### PR DESCRIPTION
**Pull request by bbfsdev**

### CODE REVIEW

File: api/handlers.go [hunk 1]
```diff
@@ -816,21 +817,26 @@ func SearchHandler(c *gin.Context) {
...
-		size,
-		preference,
-		checkTypo,
-		true,
-		true,
-		true,
-		timeoutForHighlight,
-	)
+  var res *search.QueryResult
+  if res, err = searchCache.Get(query, sortByVal, from, size); err == nil && res == nil {
```
Comment (edoshor) [inline api/handlers.go:821]: If we get an error from the cache, which is not `ErrNotFound` I think we should try elastic nonetheless. The cache is not critical.

File: search/search_cache.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func (c *SearchCache) Set(query Query, sortBy string, from int, size int, res *QueryResult) {
+  key := query.ToFullSimpleString(sortBy, from, size)
+  c.Cache.Set(key, res)
+}
+
+func MakeSearchCache() *SearchCache {
+  cache := ttlcache.NewCache()
+  cache.SetTTL(time.Duration(1 * time.Minute))
```
Comment (edoshor) [inline search/search_cache.go:34]: Can we have this larger. Maybe 5 minutes ?

File: search/search_cache.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	log "github.com/Sirupsen/logrus"
+  "github.com/jellydator/ttlcache/v2"
+)
+
+type SearchCache struct {
+  Cache ttlcache.SimpleCache
+}
+
+func (c *SearchCache) Get(query Query, sortBy string, from int, size int) (*QueryResult, error) {
+  key := query.ToFullSimpleString(sortBy, from, size)
```
Comment (edoshor) [inline search/search_cache.go:15]: Is `query.ToFullSimpleString` stable enough ?
Remember that all logically equivalent queries should be mapped to exactly the same cache key

**Review by edoshor — CHANGES_REQUESTED** [review]

**Comment by bbfsdev** [comment]
Closed due to a different design chosen. We are going to:
1. Deprecate elastic logging
2. Use nginx caching
3. Use chronicles for logging

**Pull request closed without merging**